CSharpFunctionalExtensions.Tests/ResultTests/Extensions/CheckAsyncBothTests.cs

```diff
@@ -0,0 +1,50 @@
+using FluentAssertions;
+using Xunit;
+
+namespace CSharpFunctionalExtensions.Tests.ResultTests.Extensions
+{
+    public class CheckAsyncBothTests : CheckTestsBase
+    {
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public void Tap_T_AsyncBoth_func_result(bool resultSuccess, bool funcSuccess)
```

Comment:
I think we should rename the Check / CheckIf tests accordingly, for consistency.